Sources/PointFree/Episode/Show.swift

```diff
@@ -811,19 +811,45 @@ private let exercisesView = View<[Episode.Exercise]> { exercises -> [Node] in
           ],
           ["Exercises"]
         ),
-        ol(
-          zip(1..., exercises).map {
-            li(
-              [id("exercise-\($0)")],
-              [div([markdownBlock($1.body)])]
-            )
-          }
-        )
+        ol(zip(1..., exercises).map(exercise(idx:exercise:)))
       ]
     )
   ]
 }
 
+private func exercise(idx: Int, exercise: Episode.Exercise) -> ChildOf<Tag.Ol> {
+  return li(
+    [Html.id("exercise-\(idx)")],
+    [
+      div(
+        [markdownBlock(exercise.problem)]
+          + solution(to: exercise)
+      )
+    ]
+  )
+}
+
+private func solution(to exercise: Episode.Exercise) -> [Node] {
+  guard let solution = exercise.solution else { return [] }
+
+  return [
+    details([
+      summary(["Solution"]).rawValue, // todo: update details to take children of details
```

Comment:
seems that `details` doesn't take `ChildOf<Tag.Details>`, but `summary` does return that. 

We can't fix this immediately because we still need to update to the newest swift-html in this repo, but something to come back to.

Reply:
Hm it's a bit more complicated. This is valid:

``` html
<details>
  <summary>Solution</summary>
  <p>Lorem</p>
  <p>Ipsum</p>
</details>
```

Maybe we change `details` to include a `summary:` argument and change `summary` to return a plain ole `Node` or remove the helper entirely.

``` swift
details(
  summary: "Solution",
  p("Lorem"),
  p("Ipsum")
)
```

Reply:
oh yeah, maybe we just remove the restriction on summary then